### PR TITLE
Support empty projection in ParquetRecordBatchReader

### LINKED ISSUE #1537: Support Empty Column Projection in ParquetRecordBatchReader
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

If an empty array is passed to `ArrowReader::get_record_reader_by_columns` it returns an error.

**Describe the solution you'd like**

It would be advantageous to support an empty projection within ParquetRecordBatchReader, and instead use file metadata to return the requisite number of rows. This will need the ability to create a RecordBatch with no columns and a non-zero row count #1536 

**Describe alternatives you've considered**

We could not add this functionality

**Additional context**

This would simplify the implementation of Hive based partitioning, amongst other things - https://github.com/apache/arrow-datafusion/pull/2000

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #1537

# Rationale for this change
 
See ticket

# What changes are included in this PR?

See ticket

# Are there any user-facing changes?

No, the only changes are to experimental modules.

### CODE REVIEW

File: parquet/src/arrow/arrow_reader.rs [hunk 1]
```diff
@@ -214,12 +205,6 @@ impl ParquetRecordBatchReader {
         batch_size: usize,
         array_reader: Box<dyn ArrayReader>,
     ) -> Result<Self> {
-        // Check that array reader is struct array reader
-        array_reader
```
Comment (tustvold) [inline parquet/src/arrow/arrow_reader.rs:218]: This check does not seem necessary, all it cares about is that it yields StructArray

File: parquet/src/arrow/array_reader/empty_array.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+use arrow::array::{ArrayDataBuilder, ArrayRef, StructArray};
+use arrow::datatypes::DataType as ArrowType;
+use std::any::Any;
+use std::sync::Arc;
+
+/// Returns an [`ArrayReader`] that yields [`StructArray`] with no columns
+/// but with row counts that correspond to the amount of data in the file
+///
+/// This is useful for when projection eliminates all columns within a collection
+pub fn make_empty_array_reader(row_count: usize) -> Box<dyn ArrayReader> {
```
Comment (tustvold) [inline parquet/src/arrow/array_reader/empty_array.rs:12]: I don't like this name, but it was the best I could come up with

File: parquet/src/arrow/array_reader/empty_array.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    Box::new(EmptyArrayReader::new(row_count))
+}
+
+struct EmptyArrayReader {
+    data_type: ArrowType,
```
Comment (viirya) [inline parquet/src/arrow/array_reader/empty_array.rs:34]: `data_type` seems not necessary to keep here, it is always `ArrowType::Struct(vec![])`.

**Review by viirya — APPROVED** [review]
lgtm, thanks @tustvold

File: parquet/src/arrow/array_reader/empty_array.rs (hunk 3 above)
Comment (tustvold) [reply parquet/src/arrow/array_reader/empty_array.rs:34]: It's needed for `get_data_type(&self) -> &ArrowType` as it needs to return a reference with a lifetime coupled to the object itself.

Comment (viirya) [reply parquet/src/arrow/array_reader/empty_array.rs:34]: yea, just found it and want to go to comment. you already replied. nvm.

**Review by Dandandan — APPROVED** [review]

**Pull request merged**